hack/make/validate-vet

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+source "$(dirname "$BASH_SOURCE")/.validate"
+
+IFS=$'\n'
+files=( $(validate_diff --diff-filter=ACMR --name-only -- '*.go' | grep -v '^vendor/' || true) )
+unset IFS
+
+for f in "${files[@]}"; do
+	# we use "git show" here to validate that what's committed is vetted
+    NOT_VETTED=$(git show "$VALIDATE_HEAD:$f" | go vet)
+	if [ $NOT_VETTED ]; then
+        FAILS=yes
+        echo $NOT_VETTED
+	fi
```

Comment:
Whitespace in this block is a bit inconsistent. :smile:

I'm a little concerned that this file looks so different from `validate-gofmt`, when they essentially have the same flow, so I think having them have the same structure and especially the same basic output layout would be preferable (whether that's changing this one to match that one, or changing all the `validate-*` scripts to be consistent).

Also, local script variables are traditionally `camelCase` throughout these scripts (as in `validate-gofmt`, for example).